code/modules/mob/living/simple_animal/bot/honkbot.dm

```diff
@@ -0,0 +1,380 @@
+	else
+		if(!spam_flag)
+			playsound(loc, 'sound/items/AirHorn.ogg', 100, 1, -1) //HEEEEEEEEEEEENK!!
+	sensor_blink()
+	if(spam_flag == 0)
+		if(ishuman(C))
+			C.stuttering = 20
+			C.adjustEarDamage(0, 5) //far less damage than the H.O.N.K.
+			C.Jitter(50)
+			C.Knockdown(60)
+			var/mob/living/carbon/human/H = C
+			if(client) //prevent spam from players..
+				spam_flag = TRUE
+			if (emagged <= 1) //HONK once, then leave
+				threatlevel = H.assess_threat(src)
+				threatlevel -= 6
+				target = oldtarget_name
+			else // you really don't want to hit an emagged honkbot
+				threatlevel = 6 // will never let you go
+			addtimer(CALLBACK(src, .proc/spam_flag_false), cooldowntime)
+
+			add_logs(src,C,"honked")
+
+
+			C.visible_message("<span class='danger'>[src] has honked [C]!</span>",\
+					"<span class='userdanger'>[src] has honked you!</span>")
+
+		else
+			C.stuttering = 20
+			C.Knockdown(80)
+			addtimer(CALLBACK(src, .proc/spam_flag_false), cooldowntime)
+
+
+/mob/living/simple_animal/bot/honkbot/handle_automated_action()
+	if(!..())
+		return
+
+	switch(mode)
+
+		if(BOT_IDLE)		// idle
+
+			walk_to(src,0)
+			look_for_perp()
+			if(!mode && auto_patrol)
+				mode = BOT_START_PATROL
+
+		if(BOT_HUNT)
+
+			// if can't reach perp for long enough, go idle
+			if(frustration >= 5) //gives up easier than beepsky
+				walk_to(src,0)
+				back_to_idle()
+				return
+
+			if(target)		// make sure target exists
+				if(Adjacent(target) && isturf(target.loc))
+
+					if(threatlevel <= 4)
+						honk_attack(target)
+					else
+						if(threatlevel >= 6)
+							spawn(0)
+								stun_attack(target)
+								anchored = FALSE
+								target_lastloc = target.loc
+					return
+
+				else	// not next to perp
+					var/turf/olddist = get_dist(src, target)
+					walk_to(src, target,1,4)
+					if((get_dist(src, target)) >= (olddist))
+						frustration++
+					else
+						frustration = 0
+			else
+				back_to_idle()
+
+
+		if(BOT_START_PATROL)
+			look_for_perp()
+			start_patrol()
+
+		if(BOT_PATROL)
+			look_for_perp()
+			bot_patrol()
+
+
+	return
+
+/mob/living/simple_animal/bot/honkbot/proc/back_to_idle()
+	anchored = FALSE
+	mode = BOT_IDLE
+	target = null
+	last_found = world.time
+	frustration = 0
+	INVOKE_ASYNC(src, .proc/handle_automated_action) //responds quickly
+
+/mob/living/simple_animal/bot/honkbot/proc/back_to_hunt()
+	anchored = FALSE
+	frustration = 0
+	mode = BOT_HUNT
+	INVOKE_ASYNC(src, .proc/handle_automated_action) // responds quickly
+
+/mob/living/simple_animal/bot/honkbot/proc/look_for_perp()
+	anchored = FALSE
+	for (var/mob/living/carbon/C in view(7,src))
+		if((C.stat) || (C.handcuffed))
+			continue
+
+		if((C.name == oldtarget_name) && (world.time < last_found + 100))
+			continue
+
+		threatlevel = C.assess_threat(src)
+
+		if(threatlevel <= 3)
+			if(C in view(4,src)) //keep the range short for patrolling
+				if(!spam_flag)
+					spam_flag = TRUE
```

Comment:
spam flag setting and resetting should be handled in bike_horn, if you need to skip that somewhere add a parameter.

Reply:
Fixed.